src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -306,7 +311,7 @@ private bool CanAvoidThrowingToMerger(ConcurrencyException e, int commandOffset)
                 return false;
             }
 
-            public override int Execute(DocumentsOperationContext context)
+            protected override int ExecuteCmd(DocumentsOperationContext context)
```

Comment:
Here you need to actually override `Execute`, because you need to pass the recorder to the `ParsedCommands`

Reply:
In summary, it was agreed to treat the command as one unit and not as a collection